### PR TITLE
Add blueprint for Tournament endpoints

### PR DESCRIPTION
**Pull request by williambertrand**
### Changes
- `POST /tournaments` : create a new tournament
- `GET /tournments/:id` : get tournament info
- adds some info to the readme for getting the python interpreter working for pycharm, not sure what editor you're user though Peter, but just in case some one else ever needs it

See: [Trello card](https://trello.com/c/B6D0iUUE/10-api-outline-use-blueprints-and-separate-endpoints-for-each-resource) for more info

### Testing
Responses for creating new tournaments: 
Screen Shot 2021-12-20 at 5 13 38 PM
Screen Shot 2021-12-20 at 5 13 27 PM

### CODE REVIEW

File: brackend/endpoints/tournaments/tournaments.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+class Tournaments(Resource):
+    """
+        Create a new tournament
+    """
+    def post(self):
+        body = request.get_json()
+        name = body.get("name")
+        new_tourny = save_new_tournament(name)
+        return jsonify(new_tourny)
```
Comment (williambertrand) [inline brackend/endpoints/tournaments/tournaments.py]: I think this is a good example of where a task doesn't really make sense here, we want the process to be:

1. User sends request to make a new tournament
2. We handle request, create the tourny
3. We only send back a successful response after we know the tournament has _definitely_ been created. If an error occurs, throw a 500, and the user will see an error on the app
4. Response has the new tournament info/id, so we can update the mobile app's state with the new tourny

File: poetry.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+[virtualenvs]
+in-project = true
```
Comment (williambertrand) [inline poetry.toml]: Not sure if this file should be added to .gitignore or not, I've never used poetry before 😓

File: README.md [hunk 3]
```diff
@@ -2,8 +2,16 @@
...
 
 you should get a token back, which shows you have the api up locally
 
 make sure to do `docker-compose down` and `docker-compose build` when you're making changes
+
+### Setting up PyCharm
+- install `poetry`
+- `poetry config --local virtualenvs.in-project true`
+- run `poetry install`
+- set PyCharm Python interpreter from `Preferences > Project > Python Interpreter`
```
Comment (probicheaux) [inline README.md:17]: nice, I use vim but I should probably figure out how to do the analagous thing in vim

File: brackend/endpoints/tournaments/tournaments.py (hunk 1 above)
Comment (probicheaux) [reply brackend/endpoints/tournaments/tournaments.py]: makes sense, thanks

File: poetry.toml (hunk 2 above)
Comment (probicheaux) [reply poetry.toml]: I _think_ this file is only responsible for local configuration but I'm not sure

File: brackend/app.py [hunk 4]
```diff
@@ -17,9 +17,20 @@
 )
 from brackend.util import BrackendException
 
+# App Blueprints: Blueprint for each resource
+from endpoints.tournaments.tournaments import tournament_bp
+
+api_prefix_v1 = "/api/v1"
+def register_blueprints(app):
```
Comment (probicheaux) [inline brackend/app.py:24]: dope

**Review by probicheaux — APPROVED** [review]
dope, good to know about resources/blueprints

**Pull request merged**